aten/src/ATen/core/ivalue_inl.h

```diff
@@ -377,7 +377,7 @@ struct C10_EXPORT ivalue::Future : c10::intrusive_ptr_target {
   }
 
   // Get the result of the current future.
-  IValue value() {
```

Comment:
Are IValues immutable? Do they have value or pointer semantics?

My worry is that, if this function effectively returns a new _pointer_ to the same value, then the caller could modify it and thus (in some sense) modify the future even if it only had const access to it.

Reply:
I see that below constValue is being replaced by value in order to call toCustomClass on it. It seems toCustomClass should be able to operate on const& (there's a specific overload qualified as such). If that is the reason for this change, I'd investigate why toCustomClass doesn't work as intended...

Reply:
Yep, constValue works, let me update.